LDAR_Sim/model_code/input_manager.py

```diff
@@ -0,0 +1,216 @@
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the MIT License as published
+# by the Free Software Foundation, version 3.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# MIT License for more details.
+
+# You should have received a copy of the MIT License
+# along with this program.  If not, see <https://opensource.org/licenses/MIT>.
+#
+# ------------------------------------------------------------------------------
+
+import copy
+import os
+import yaml
+import json
+import sys
+
+from utils.check_parameter_types import check_types
+from default_parameters.default_global_parameters import default_global_parameters
+from default_parameters.default_program_parameters import default_program_parameters
+from default_parameters.default_aircraft_parameters import default_aircraft_parameters
+from default_parameters.default_OGI_parameters import default_OGI_parameters
+from default_parameters.default_OGI_FU_parameters import default_OGI_FU_parameters
+from default_parameters.default_satellite_parameters import default_satellite_parameters
+from default_parameters.default_truck_parameters import default_truck_parameters
+from default_parameters.default_continuous_parameters import default_continuous_parameters
+
+
+class InputManager:
+    def __init__(self):
+        """ Constructor creates a lookup of method defaults to run validation against
+        """
+        # Simulation parameter are parameters that are set up for this simulation
+        self.simulation_parameters = copy.deepcopy(default_global_parameters)
+
+        # Construct a lookup of default method parameters
+        self.method_defaults = {
+            'aircraft': copy.deepcopy(default_aircraft_parameters),
+            'OGI': copy.deepcopy(default_OGI_parameters),
+            'OGI_FU': copy.deepcopy(default_OGI_FU_parameters),
+            'satellite': copy.deepcopy(default_satellite_parameters),
+            'truck': copy.deepcopy(default_truck_parameters),
+            'continuous': copy.deepcopy(default_continuous_parameters),
+        }
+        return
+
+    def read_and_validate_parameters(self, parameter_filenames):
+        """Method to read and validate parameters
+        :param parameter_filenames: a list of paths to parameter files
+        :return returns fully validated parameters dictionary for simulation in LDAR-Sim
+        """
+        raw_parameters = self.read_parameter_files(parameter_filenames)
+        self.parse_parameters(raw_parameters)
+
+        # Coerce all paths to absolute paths prior to release, add extra guards for other parts of the code
+        # that concatenate strings to construct file paths, and expect trailing slashes
+        self.simulation_parameters['wd'] = os.path.abspath(self.simulation_parameters['wd']) + '//'
```

Comment:
WE probably should be using the pathlib library and absolute imports in all cases, that way it can be called from any folder with any operating system.

`from pathlib import Path`
```
self.src_dir = Path(os.path.dirname(os.path.realpath(__file__)))
self.root_dir = self.src_dir.parent
self.output_directory = self.root_dir / "outputs/"
```

Reply:
Full path fix for entire model required.